### PR TITLE
fix flaky hdfs test

### PR DESCRIPTION
**Pull request by chaudum**
@mfussenegger please review
jenkins ran 100 iterations without failure. so I assume it's fixed :)

### CODE REVIEW

File: blackbox/hdfs/src/tests.py [hunk 1]
```diff
@@ -100,26 +102,48 @@ def _get_hadoop(self):
...
             env={
                 'JAVA_HOME': JAVA_HOME,
                 'HADOOP_CLASSPATH': self.yarn_server_jar
-            })
-        elapsed = 0.0
-        while not is_up('localhost', int(NN_PORT)) and elapsed < 5:
-            time.sleep(0.1)
-            elapsed += 0.1
+            },
+            stdout=subprocess.PIPE,
```
Comment (mfussenegger) [inline blackbox/hdfs/src/tests.py]: Could this cause the PIPE to get full and the process get stuck if there is a lot of output?

File: blackbox/hdfs/src/tests.py [hunk 2]
```diff
@@ -100,26 +102,48 @@ def _get_hadoop(self):
...
+            '-nnport', NN_PORT, '-nomr', '-format',
+            '-D', 'dfs.replication=0',
+            '-D', 'dfs.client.use.datanode.hostname=true',
+            '-D', 'dfs.datanode.use.datanode.hostname=true',
+        ]
 
         JAVA_HOME = os.environ.get('JAVA_HOME', '/usr/lib/jvm/java-8-openjdk/')
+
+        print('')
+        print('>>> ' + 'JAVA_HOME=' + JAVA_HOME)
```
Comment (mfussenegger) [inline blackbox/hdfs/src/tests.py]: were these for debugging purpose or is it okay to keep them?

File: blackbox/hdfs/src/tests.py (hunk 1 above)
Comment (chaudum) [inline blackbox/hdfs/src/tests.py]: there is actually only a single line printed to `stdout`

File: blackbox/hdfs/src/tests.py (hunk 2 above)
Comment (chaudum) [inline blackbox/hdfs/src/tests.py]: I can remove them. Guess they are not *that* interesting any more once it's stable.

**Pull request merged**